htdocs/core/class/html.form.class.php

```diff
@@ -2542,6 +2542,10 @@ public function select_produits_list($selected = '', $htmlname = 'productid', $f
 		global $langs, $conf;
 		global $hookmanager;
 
+		if (empty($hookmanager->hooks)) {
+			$hookmanager->initHooks(array('commonobject'));
```

Comment:
context of hook must be initialized by the parent page or by the parent CLI script so context is the same from the start until the end of code.
We must avoid init contact on low level functions.

Reply:
As I recall, every card worked as expected except for offers (propale).  Upon examining the `$hookmanager` that I received from the propale card, there was a context, but the hook was empty which is why I ran `initHooks` when the hook in `$hookmanager` was empty.  I bases the above code on code that I found in the file.

Late next week, I can look at why the propale card is not giving me a hook whereupon I can fix it and remove the above could if you would prefer that solution.

Reply:
Yes, please, we must find why a proposal card doe not set the hook when you call the select_produits_list().
It should.